### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by Poddyman0**

### CODE REVIEW

File: src/components/ChangeVotes.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    patchVotes(vote);
+    console.log(vote, "line 18")
+}
+
+const patchVotes = (vote) => {
+    axios
+    .patch(`https://be-nc-news-tnfa.onrender.com/api/articles/${article_id}`)
+    .then({data})
+    console.log(data, "line 25")
+    setAlterVotes(data.article, { inc_votes : vote})
```
Comment (Hard-Luck) [inline src/components/ChangeVotes.jsx]: You would want to change the votes first for optimistic rendering and revert it in the case of an error

File: src/components/ChangeVotes.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+//optimistic rendering below
+const decreaseVotes = (vote) => {
+    patchVotes(vote);
+    console.log(vote, "line 18")
+}
+
+const patchVotes = (vote) => {
+    axios
+    .patch(`https://be-nc-news-tnfa.onrender.com/api/articles/${article_id}`)
+    .then({data})
```
Comment (Hard-Luck) [inline src/components/ChangeVotes.jsx]: This should be an arrow function wrapping the rest of the content

File: src/components/ChangeVotes.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    console.log(data, "line 25")
+    setAlterVotes(data.article, { inc_votes : vote})
+}
+  return (
+    <>
+    <button onClick={increaseVotes(+1)}>Add vote</button>
```
Comment (Hard-Luck) [inline src/components/ChangeVotes.jsx]: here the increaseVotes function is being invoked immedaitely, I think you want to to be invoked when cliicked so look at wrapping it in a function ()=> increaseVotes(1)

File: src/components/ChangeVotes.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+}
+//optimistic rendering below
+const decreaseVotes = (vote) => {
+    patchVotes(vote);
+    console.log(vote, "line 18")
+}
+
+const patchVotes = (vote) => {
+    axios
+    .patch(`https://be-nc-news-tnfa.onrender.com/api/articles/${article_id}`)
```
Comment (Hard-Luck) [inline src/components/ChangeVotes.jsx]: This is giving an article_id does not exist, it might be worth looking at your back end

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
Lots of console.logs need removing too

**Review by Pipe-Smoking-Rabbit — COMMENTED** [review]
Nice work on these tickets, here is some feedback from a user perspective:
- The landing / home page doesn't load any content, does there need to be a home page and an _All Articles_ page separately ? ( You might have future plans for the home page which is fine too )
- The Article List shows me the article ID for each one, as a user do I need to see that information ?
- The single article page doesn't show me the article body so as a user I can't read the article 
- The single article page has a label for _"Comment Count:"_ but it isn't currently showing the number
- The comments list renders nicely underneath the article page which feels like a good place for it to be 👍 
- As a user do I need to see the comment ID for each comment ?

**Pull request merged**